src/reformatters/u_arizona/swann/validators.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from reformatters.common import validation
+
+# For regions outside of CONUS, the values in this dataset are expected
+# to be NaNs. We sampled various times across the dataset and determined
+# the expected number of NaNs to be ~46.425% of the data.
+EXPECTED_NAN_PERCENTAGE = 46.425
+MAX_NAN_PERCENTAGE = EXPECTED_NAN_PERCENTAGE + 0.001
+
+
+def check_data_is_current(ds: xr.Dataset) -> validation.ValidationResult:
+    """
+    Check that the data is current within the last 48 hours.
+    """
+    now = pd.Timestamp.now()
+    latest_init_time_ds = ds.sel(time=slice(now - pd.Timedelta(hours=48), None))
+    if latest_init_time_ds.sizes["time"] == 0:
+        return validation.ValidationResult(
+            passed=False, message="No data found for the last 48 hours"
+        )
+
+    return validation.ValidationResult(
+        passed=True,
+        message="Data found for the last 48 hours",
+    )
+
+
+def check_latest_time_nans(ds: xr.Dataset) -> validation.ValidationResult:
+    """
+    Check that the data does not have more than the expected proportion of NaNs
+    for the latest time step.
+    """
+    sample_ds = ds.isel(time=-1)
+    return _check_nans_in_ds("check_latest_time_nans", sample_ds)
+
+
+def check_random_time_within_last_year_nans(
+    ds: xr.Dataset,
+) -> validation.ValidationResult:
+    """
+    Check that the data does not have more than the expected proportion of NaNs
+    for a random time in the last year, as we pull a years worth of data for each operational
+    update of the dataset.
+    """
+    random_time_index = np.random.choice(365)
+    sample_ds = ds.isel(time=slice(random_time_index, random_time_index + 1))
```

Comment:
I think this will only select from the first year of the dataset?
```suggestion
    random_time_index = np.random.choice(365) + 1
    sample_ds = ds.isel(time=-random_time_index)
```

Reply:
Oops nice catch :pray: